vnext/ReactUWP/Views/TouchEventHandler.cpp

```diff
@@ -88,7 +88,11 @@ void TouchEventHandler::OnPointerPressed(
     UpdatePointersInViews(instance, args, tag, sourceElement);
 
     size_t pointerIndex = AddReactPointer(args, tag, sourceElement);
-    DispatchTouchEvent(TouchEventType::Start, pointerIndex);
+
+    // For now, we only want to send click events for the left mouse button.
+    if (m_pointers[pointerIndex].isLeftButton) {
```

Comment:
Will this block finger taps?

Reply:
This says that finger taps also produce the PointerPressed event 
(https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.uielement.pointerpressed?view=winrt-19041#remarks)

Reply:
Sure PointerPressed will be raised, but this code is explicitly checking the pointer to see if the left button, which I believe will not be the case for a touch event.

You can test touch on a non-touch device with a little utility app called MouseHookApp.  It converts mouse into touch.
Go here:
\\kmelmondev\public\MouseHookApp.exe
Once launched all mouse is converted to touch.  Kill the app to go back to normal.

Can you test out your change with MouseHookApp?


---
In reply to: [426197483](https://github.com/microsoft/react-native-windows/pull/4878#discussion_r426197483) [](ancestors = 426197483)

Reply:
@kmelmon I've run your app, and the resulting touch events are still interpreted as left mouse button clicks:
![image](https://user-images.githubusercontent.com/12816515/82363116-3fb8c400-99c2-11ea-9bf0-c19dd0e8d386.png)

Reply:
Interesting!  I wonder if the mouse button information is being preserved.  What happens if you use the right mouse button with MouseHookApp?

Reply:
@kmelmon 
![image](https://user-images.githubusercontent.com/12816515/82512349-af16dc80-9ac4-11ea-9644-9e434beafc59.png)
Device type is 'Mouse' not touch in this case, but button info is preserved.
Same happens if I press the wheel (middle button)

Reply:
I think that means that MouseHookApp isn't truly converting to pointer, and Kai's concern seems valid.  Sorry about that! 

I think you'll also need to query the deviceType, ie
if deviceType == pointer => let it through
if deviceType == mouse => let it through only if isLeftButton is true

Reply:
I've just tried RD from my Surface, and tapping elements gives the same result as a left mouse click using the MouseHook app - deviceType=Touch, isLeftButton=True

Reply:
Discussed offline.  With the help of Ross Nichols from the input team, we determined that all the input types we care about including pen will supply isLeftonButton == True.  This change looks good to me.

---
In reply to: [428833625](https://github.com/microsoft/react-native-windows/pull/4878#discussion_r428833625) [](ancestors = 428833625)